Basic model on DFR similarity documents options no longer accepted

https://www.elastic.co/guide/en/elasticsearch/reference/7.0/index-modules-similarity.html

`basic_model` still documents the following options:: `be, d, g, if, in, ine and p.` However atleast `d` is no longer supported. 

The docs mention:

> be, d and p should be avoided in practice as they might return scores that are equal to 0 or infinite with terms that do not meet the expected random distribution.

But it seems these options can no longer be used

```json
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "Basic model [d] isn't supported anymore, please use another model.",
        "stack_trace" : "[Basic model [d] isn't supported anymore, please use another model.]; nested: IllegalArgumentException[Basic model [d] isn't supported anymore, please use another model.];\r\n\tat org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:639)\r\n\tat org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:567)\r\n\tat org.elasticsearch.rest.BytesRestResponse.build(BytesRestResponse.java:138)\r\n\tat org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:96)\r\n\tat org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:91)\r\n\tat org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:58)\r\n\tat org.elasticsearch.action.support.TransportAction$1.onFailure(TransportAction.java:74)\r\n\tat org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction$1.onFailure(TransportMasterNodeAction.java:201)\r\n\tat org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:69)\r\n\tat org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:69)\r\n\tat org.elasticsearch.cluster.AckedClusterStateUpdateTask.onFailure(AckedClusterStateUpdateTask.java:79)\r\n\tat org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$IndexCreationTask.onFailure(MetaDataCreateIndexService.java:612)\r\n\tat org.elasticsearch.cluster.service.MasterService$SafeClusterStateTaskListener.onFailure(MasterService.java:499)\r\n\tat org.elasticsearch.cluster.service.MasterService$TaskOutputs.notifyFailedTasks(MasterService.java:432)\r\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:211)\r\n\tat org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:142)\r\n\tat org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)\r\n\tat org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:681)\r\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)\r\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\r\n\tat java.base/java.lang.Thread.run(Thread.java:834)\r\nCaused by: java.lang.IllegalArgumentException: Basic model [d] isn't supported anymore, please use another model.\r\n\tat org.elasticsearch.index.similarity.SimilarityProviders.parseBasicModel(SimilarityProviders.java:143)\r\n\tat org.elasticsearch.index.similarity.SimilarityProviders.createDfrSimilarity(SimilarityProviders.java:306)\r\n\tat org.elasticsearch.index.similarity.SimilarityService.lambda$static$10(SimilarityService.java:104)\r\n\tat org.elasticsearch.index.similarity.SimilarityService.<init>(SimilarityService.java:141)\r\n\tat org.elasticsearch.index.IndexModule.newIndexService(IndexModule.java:398)\r\n\tat org.elasticsearch.indices.IndicesService.createIndexService(IndicesService.java:518)\r\n\tat org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:467)\r\n\tat org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$IndexCreationTask.execute(MetaDataCreateIndexService.java:471)\r\n\tat org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:47)\r\n\tat org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:687)\r\n\tat org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:310)\r\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:210)\r\n\t... 9 more\r\n"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "Basic model [d] isn't supported anymore, please use another model.",
    "stack_trace" : "java.lang.IllegalArgumentException: Basic model [d] isn't supported anymore, please use another model.\r\n\tat org.elasticsearch.index.similarity.SimilarityProviders.parseBasicModel(SimilarityProviders.java:143)\r\n\tat org.elasticsearch.index.similarity.SimilarityProviders.createDfrSimilarity(SimilarityProviders.java:306)\r\n\tat org.elasticsearch.index.similarity.SimilarityService.lambda$static$10(SimilarityService.java:104)\r\n\tat org.elasticsearch.index.similarity.SimilarityService.<init>(SimilarityService.java:141)\r\n\tat org.elasticsearch.index.IndexModule.newIndexService(IndexModule.java:398)\r\n\tat org.elasticsearch.indices.IndicesService.createIndexService(IndicesService.java:518)\r\n\tat org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:467)\r\n\tat org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$IndexCreationTask.execute(MetaDataCreateIndexService.java:471)\r\n\tat org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:47)\r\n\tat org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:687)\r\n\tat org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:310)\r\n\tat org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:210)\r\n\tat org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:142)\r\n\tat org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)\r\n\tat org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)\r\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:681)\r\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252)\r\n\tat org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\r\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\r\n\tat java.base/java.lang.Thread.run(Thread.java:834)\r\n"
  },
  "status" : 400
}
```

--- a/docs/reference/index-modules/similarity.asciidoc
+++ b/docs/reference/index-modules/similarity.asciidoc
@@ -92,22 +92,14 @@ from randomness] framework. This similarity has the following options:
 
 [horizontal]
 `basic_model`::
-    Possible values: {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelG.html[`be`],
-    {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelD.html[`d`],
-    {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelG.html[`g`],
+    Possible values: {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelG.html[`g`],
     {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelIF.html[`if`],
-    {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelIn.html[`in`],
-    {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelIne.html[`ine`] and
-    {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelP.html[`p`].
-
-`be`, `d` and `p` should be avoided in practice as they might return scores that
-are equal to 0 or infinite with terms that do not meet the expected random
-distribution.
+    {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelIn.html[`in`] and
+    {lucene-core-javadoc}/org/apache/lucene/search/similarities/BasicModelIne.html[`ine`].
 
 `after_effect`::
-    Possible values: {lucene-core-javadoc}/org/apache/lucene/search/similarities/AfterEffect.NoAfterEffect.html[`no`],
-    {lucene-core-javadoc}/org/apache/lucene/search/similarities/AfterEffectB.html[`b`] and
-    {lucene-core-javadoc}/org/apache/lucene/search/similarities/AfterEffectL.html[`l`].
+    Possible values: {lucene-core-javadoc}/org/apache/lucene/search/similarities/AfterEffectB.html[`b`] and
+    {lucene-core-javadoc}/org/apache/lucene/search/similarities/AfterEffectB.html[`l`].
 
 `normalization`::
     Possible values: {lucene-core-javadoc}/org/apache/lucene/search/similarities/Normalization.NoNormalization.html[`no`],

